EcoSoapBank/Network Controller/GraphQLController.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  GraphQLController.swift
+//  EcoSoapBank
+//
+//  Created by Christopher Devito on 8/10/20.
+//  Copyright © 2020 Spencer Curtis. All rights reserved.
+//
+
+import Foundation
+
+enum HTTPMethod: String {
+    case post = "POST"
+}
+
+/// Class containing methods for communicating with GraphQL backend
+class GraphQLController {
+
+    // MARK: - Properties
+
+    private let url = URL(string: "http://35.208.9.187:9094/ios-api-1/")!
+    // Setting up the url request
+    private lazy var request: URLRequest = {
+        var request = URLRequest(url: url)
+        request.httpMethod = HTTPMethod.post.rawValue
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        return request
+    }()
+
+    // MARK: - Request methods
+
+    /// Method for GraphQL query requests
+    /// - Parameters:
+    ///   - query: The intended query in string format
+    ///   - session: The URLSession used for the request. By default this is URLSession.shared
+    ///   - completion: Completion handler that passes back a Result of type Profile or Error
+    func queryRequest(query: String,
+                      session: URLSession = URLSession.shared,
+                      completion: @escaping (Result<Profile, Error>) -> Void) {
```

Comment:
Yeah, this will be a little tricky to make generic in a way that's "safe" and effective... one thing you could do is pass the type into the method directly, so it'd look something like this:

```swift
func queryRequest<Type>(
    query: String,
    for type: Type.type,
    completion: @escaping (Result<Type, Error>) -> Void
)
```

I'm not sure how well that will work with JSON serialization. Since we can't use queries for this project yet it might be a good idea to try something like that with one of the projects you were playing around with for testing.